src/main/java/io/wispforest/owo/ui/core/Size.java

```diff
@@ -31,4 +36,12 @@ public static Size square(int sideLength) {
     public static Size zero() {
         return ZERO;
     }
+
+    public static StructEndec<Size> createEndec(String widthName, String heightName) {
+        return StructEndecBuilder.of(
+                Endec.INT.fieldOf(widthName, Size::width),
+                Endec.INT.fieldOf(heightName, Size::height),
+                Size::of
+        );
+    }
```

Comment:
is this method really that useful?
i know you're using it in NinePatchTexture but imo making an endec for Size with different keys is niche enough that it can either do it itself or use an internal method

Reply:
Like I guess